src/net/packet.rs

```diff
@@ -6800,3 +6808,91 @@ pub struct sP_FE2LS_DISCONNECTING {
     pub iTempValue: i32,
 }
 impl FFPacket for sP_FE2LS_DISCONNECTING {}
+
+#[repr(packed(4))]
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct sP_FE2LS_REQ_SEND_BUDDY_FREECHAT {
+    pub iFromPCUID: i64,
+    pub iToPCUID: i64,
+    pub szFreeChat: [u16; 128],
+    pub iEmoteCode: i32,
+}
+impl FFPacket for sP_FE2LS_REQ_SEND_BUDDY_FREECHAT {}
+
+#[repr(packed(4))]
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+pub struct sP_LS2FE_REP_SEND_BUDDY_FREECHAT {
```

Comment:
This packet should definitely be `REQ` instead of `REP` since the login server is **requesting** that the message be sent to the player; you'll need to update in a few spots